### DISCUSSION TITLE
Install extensions specified in sync settings automatically

### DISCUSSION THREAD

**Discussion by iloveitaly** [community/community#33832] [Codespaces] [answered]
Hello! It looks like the codespace is not installed the extensions in my settings sync automatically. I need to manually click the extensions I want to install:

<img width="367" alt="image" src="https://user-images.githubusercontent.com/150855/191756019-d2782777-ccdb-4364-9d53-bdd9f75973ee.png">

Is there a way to automatically install these extensions when the codespace is created?

**Comment by samruddhikhandale** [comment]
Hi, Could you try reloading the Window and see if the synced extensions get installed?

You could use the `Reload Window` command from the command palette 

![image](https://user-images.githubusercontent.com/24955913/191854856-b0aabe47-fde6-4299-aed0-7346780b8258.png)

**Comment by iloveitaly** [comment]
This did not work for me. Going to try on a fresh container and see if I get a different result.

Do you know if extensions do not install from settings sync if you specify a list of extensions to install in the `devcontainer.json`?

**Comment by samruddhikhandale** [comment]
Let me know if it works in a new codespace.

> Do you know if extensions do not install from settings sync if you specify a list of extensions to install in the devcontainer.json?

I tried to reproduce, and it worked for me. I can see my synced extensions installed along with the dev container config.

![image](https://user-images.githubusercontent.com/24955913/192010010-a39006ce-12a9-49ae-b084-daf3c118dfaa.png)

I have few questions, and wondering why I am not able to reproduce

1. Are you creating the codespaces in web or desktop? 
2. If you are using desktop, can you make sure your VS Code and Codespaces extension is up to date? 
3. Also, can you send us the settings sync logs and try the `Settings Sync: Sync now` command ? 

![image](https://user-images.githubusercontent.com/24955913/192010932-83561ac9-9ec6-4a91-9f7c-c04a81e0343f.png)

**Reply by iloveitaly** [reply]
@samruddhikhandale it's not working for me. [`code --install-extension` works fine for me.](https://github.com/iloveitaly/dotfiles/blob/bdbdf188802c00935736b51468eb36ad9fa9214b/codespace.sh#L26).

Here's my `devcontainer.json`:

```
{
	"name": "Ubuntu",
	"build": {
		"dockerfile": "Dockerfile",
		// Update 'VARIANT' to pick an Ubuntu version: jammy / ubuntu-22.04, focal / ubuntu-20.04, bionic /ubuntu-18.04
		// Use ubuntu-22.04 or ubuntu-18.04 on local arm64/Apple Silicon.
		"args": { "VARIANT": "ubuntu-22.04" }
	},

	// Use 'forwardPorts' to make a list of ports inside the container available locally.
	// "forwardPorts": [],

	// Use 'postCreateCommand' to run commands after the container is created.
	"postCreateCommand": "sh ./.devcontainer/post-install.sh",

	// Comment out to connect as root instead. More info: https://aka.ms/vscode-remote/containers/non-root.
	"remoteUser": "vscode",
	"features": {
		"homebrew": "latest"
	}
}
```

So strange! I've tried reloading the window, running sync now, etc. Show sync settings indicates that extensions are synced:

<img width="209" alt="image" src="https://user-images.githubusercontent.com/150855/192100253-69e02cbc-9a8c-4344-88ba-e12a9256b5fb.png">

**Comment by softwaredevelop** [comment, marked as answer]
Hello,

add to VS Code settings.json:

```
  "github.codespaces.defaultExtensions": [
    "extension.1",
    "extension.2",
    "extension.3",
    "extension...",
  ]
```

After modified VS Code settings.json rebuild your codspace.

**Reply by iloveitaly** [reply]
Yup, that works, but what I'm looking to do is have extensions automatically installed—without having to specify them—from my settings sync extension list. Extensions specified in the devcontainer.json are working, but the settings sync extension install seems broken.

**Reply by softwaredevelop** [reply]
this is the way
you can add default codespace extensions in user settings.json
![2022-09-30s](https://user-images.githubusercontent.com/61334390/193155915-9011b566-a8e0-4e57-9232-6cd1132d80d2.png)

**Reply by iloveitaly** [reply]
This fixed it! I did not notice the `codespaces` in the settings page you specified. Thank you!
